### PR TITLE
Write correct Iceberg types to Glue metastore

### PR DESCRIPTION
**Pull request by findepi**

### CODE REVIEW

**Comment by findepi** [comment]
Note that Iceberg own GlueCatalog has similar problem:
https://github.com/apache/iceberg/blob/370c135144ce2d1b00e9938e267548e78ee6edaf/aws/src/main/java/org/apache/iceberg/aws/glue/IcebergToGlueConverter.java#L263-L268
 
Relevant discussion on iceberg slack: https://apache-iceberg.slack.com/archives/C03LG1D563F/p1689351883036489

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/Column.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+
+@Immutable
```
Comment (alexjo2144) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/Column.java:30]: Are we annotating immutable classes now? That's new to me

**Review by alexjo2144 — COMMENTED** [review]
I didn't think we wrote any column information for Iceberg tables to Glue, am I wrong about that?

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/Column.java (hunk 1 above)
Comment (findepi) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/Column.java:30]: We sometimes do. I think this annotation is generally desirable. This class is a copy of other one, and i just copied annotation anyway.

**Comment by findepi** [comment]
> I didn't think we wrote any column information for Iceberg tables to Glue, am I wrong about that?

you're right. I didn't finish the work.

**Comment by findepi** [comment]
Superseded by https://github.com/trinodb/trino/pull/18315

**Pull request closed without merging**